OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs

```diff
@@ -98,13 +98,16 @@ public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 				.SelectMany(p => p.Render());
 		}
 
-		public EditorActorPreview Add(ActorReference reference) { return Add(NextActorName(), reference); }
+		public void Add(ActorReference reference) { Add(NextActorName(), reference); }
 
-		EditorActorPreview Add(string id, ActorReference reference, bool initialSetup = false)
+		void Add(string id, ActorReference reference, bool initialSetup = false)
```

Comment:
Making this `void` breaks the symmetry with `Remove`, and means you need to do gymnastics if you want to add an actor and then immediately do something with the `EditorActorPreview`.

Can you instead just skip the screenMap addition on empty bounds?